src/components/App/App.jsx

```diff
@@ -1,26 +1,40 @@
-import { useState, useEffect } from 'react';
+import { useState } from 'react';
 import { Routes, Route } from 'react-router-dom';
-import './App.css';
-import Home from '../Home/Home.jsx';
+import Form from '../Form/Form';
 import SavedCards from '../SavedCards/SavedCards.jsx';
+import EditCaption from '../EditCaption/EditCaption.jsx';
 import Error from '../Error404/Error404.jsx';
 
 function App() {
   const [cards, setCards] = useState([]);
+  const [errorMessage, setErrorMessage] = useState('');
 
-  const addCard = (card) => {
-    setCards((prevCards) => [...prevCards, card]);
+  const addCard = (newCard) => {
+    const exists = cards.some(card => card.image === newCard.image);
+    if (!exists) {
+      setCards((prevCards) => [...prevCards, newCard]);
+      setErrorMessage('');
+    } else {
+      setErrorMessage('Cannot add duplicate image.');
+    }
+  };
+
+  const onSave = (imageSrc, newCaption) => {
+    const updatedCards = cards.map(card => {
+      if (card.image === imageSrc) {
+        return { ...card, caption: newCaption };
+      }
+      return card;
+    });
+    setCards(updatedCards);
   };
 
-  useEffect(() => {
-    console.log(cards[0]);
-  }, [cards]);
-  
   return (
     <main className='App'>
       <Routes>
-        <Route path='/' element={<Home addCard={addCard} />} />
```

Comment:
Nice that you cleaned up code by consolidating your Home component and Form components.